lib/easyredmine_budget_quotas/time_entry_validation.rb

```diff
@@ -71,41 +74,45 @@ def check_if_budget_quota_valid
 
             assignable_hours = assignable_value/value_per_hour
 
-            # get current indx from comment
-            comment_id = self.comments.match(/(?<=\[)[0-9]{1,}/)
+            # get current index from comment
+            comment_id = self.comments.match(/(?<=\[)[0-9]{1,}/).to_id rescue nil
             if comment_id.nil?
               self.comments = "[1] #{self.comments}"
             else
               self.comments = "[#{comment_id.to_i+1}] #{self.comments.gsub(/\[[0-9]{1,}\]\ /, '')}"
             end
 
-
             # store values for next time entry and close current BudgetQuota
             @remaining_values_for_assignment = self.attributes.merge('hours' => self.hours - assignable_hours)
             current_bqs.first.update_column(:budget_quota_exceeded, true)
```

Comment:
@florianeck 
this, is party working but really has to be fixed, because if somebody changes (remove / lower the hours) a non locked entry, the remaining restbudget is than never used